### PR TITLE
Update docs for supabase-js v2 GA

### PR DESCRIPTION
**Pull request by dannykng**
## What kind of change does this PR introduce?

- [x] Update version from `v2 RC` to `v2`
- [x] Remove `/next` from URLs
- [x] Update callouts on supabase-js reference docs landing pages
- [x] Update guides using `signIn()` to `signInWithPassword()`, `signInWithOAuth()`, or `signInWithOtp()`
- [x] Update mentions of `npm install @supabase/supabase-js@rc` to `npm install @supabase/supabase-js`
- [x] Update mentions of `auth.session()` to `auth.getSession()`
- [ ] Update quickstarts

## Additional context

Doc | Before | After
-- | --| --
supabase-js v2 ref docs landing page | Screen Shot 2022-10-06 at 3 21 31 PM | Screen Shot 2022-10-06 at 3 21 18 PM
supabase-js v1 ref docs landing page | Screen Shot 2022-10-06 at 3 23 14 PM | Screen Shot 2022-10-06 at 3 23 34 PM
signIn() -> signInWithPassword() | Screen Shot 2022-10-06 at 3 45 43 PM | Screen Shot 2022-10-06 at 3 45 58 PM
signIn() -> signInWithOtp() | Screen Shot 2022-10-06 at 3 54 48 PM | Screen Shot 2022-10-06 at 3 55 00 PM
signIn() -> signInWithOAuth() | Screen Shot 2022-10-06 at 3 25 10 PM | Screen Shot 2022-10-06 at 3 25 24 PM

### CODE REVIEW

File: apps/reference/docs/guides/auth.mdx [hunk 1]
```diff
@@ -177,7 +177,14 @@ Supabase provides a special function in Postgres, `auth.uid()`, which extracts t
 
 ## User Management
 
-Supabase makes it simple to manage your users.
+Supabase provides multiple endpoints to authenticate and manage your users:
+- [Sign up](/docs/reference/javascript/auth-signup)
+- [Sign in with password](/docs/reference/javascript/auth-signinwithpassword)
+- [Sign in with one-time password (OTP)](/docs/reference/javascript/auth-signinwithotp)
```
Comment (kangmingtay) [inline apps/reference/docs/guides/auth.mdx:183]:
```suggestion
- [Sign in with passwordless / one-time password (OTP)](/docs/reference/javascript/auth-signinwithotp)
```

File: apps/reference/docs/guides/auth/auth-apple.mdx [hunk 2]
```diff
@@ -58,17 +56,17 @@ This will serve as the `client_id` when you make API calls to authenticate the u
   - Click `Continue` at the top right.
   - Click `Register` at the top right.
 
-### Find your callback URL
+## Find your callback URL
 
 The next step requires a callback URL, which looks like this:
 
-`https://<project-ref>.supabase.co/auth/v1/callback`
+`https://<project-ref>.supabase.co/auth/v2/callback`
```
Comment (kangmingtay) [inline apps/reference/docs/guides/auth/auth-apple.mdx:63]: i don't think we're changing the version number in the url

File: apps/reference/docs/guides/auth/auth-apple.mdx [hunk 3]
```diff
@@ -58,17 +56,17 @@ This will serve as the `client_id` when you make API calls to authenticate the u
...
 
 - Go to your [Supabase Project Dashboard](https://app.supabase.com).
 - Click on the `Settings` icon at the bottom of the left sidebar.
 - Click on `API` in the list.
 - Under Config / URL you'll find your API URL, you can click `Copy` to copy it to the clipboard.
-- Now just add `/auth/v1/callback` to the end of that to get your full `OAuth Redirect URI`.
+- Now just add `/auth/v2/callback` to the end of that to get your full `OAuth Redirect URI`.
```
Comment (kangmingtay) [inline apps/reference/docs/guides/auth/auth-apple.mdx:69]: ditto above

**Review by kangmingtay — DISMISSED** [review]

**Review by alaister — APPROVED** [review]

**Review by kiwicopple — APPROVED** [review]

**Review by kiwicopple — APPROVED** [review]
let's gooooo

**Pull request merged**